swan-lake/integration-tutorials/revamped-sending-a-message-to-a-service.md

```diff
@@ -0,0 +1,370 @@
+---
+layout: ballerina-sending-a-message-to-a-service-left-nav-pages-swanlake
+title: Sending a message to a service 
+permalink: /learn/sending-a-message-to-a-service/
+description: Integration tutorial for sending a message to a service.
+keywords: ballerina, programming language, client, restful-api, integration
+active: sending-a-message-to-a-service
+intro: This tutorial helps you understand the basics of how Ballerina can be used to do client calls and develop RESTful APIs.
+---
+
+## Overview
+
+In this tutorial, you will develop a service that allows a user to retrieve a list of doctors based on the doctor's specialization (category). The information about the doctors is retrieved from a separate microservice. 
+
+To implement this use case, you will develop a REST service with a single resource using Visual Studio Code with the Ballerina Swan Lake extension, and then, run the service. The resource will receive the user request, retrieve details from the backend service, and respond to the user request with the relevant doctor details.
+
+### Concepts covered
+
+- REST API
+- HTTP client
+
+## Develop the application
+
+### Step 1: Set up the workspace
+
+Install [Ballerina Swan Lake](https://ballerina.io/downloads/) and the [Ballerina Swan Lake extension](https://marketplace.visualstudio.com/items?itemName=wso2.ballerina) on VS Code.
+
+### Step 2: Develop the service
+
+Follow the instructions given in this section to develop the service.
+
+1. Create a new Ballerina project using the `bal` command and open it in VS Code.
+
+    ```bash
+    $ bal new sending-a-message-to-a-service
+    ```
+2. Remove the example content and open Ballerina diagram view with [Ballerina HTTP API Designer](https://wso2.com/ballerina/vscode/docs/design-the-services/http-api-designer) in VS Code.
+
+    ![Open diagram view](/learn/images/tutorial_sending_a_message_to_a_service_open_diagram_view.gif)
+
+3. Define two [configurable variables](https://ballerina.io/learn/by-example/#configurability) for the port on which the listener should listen and the URL of the backend service.
+
+    ![Define configurable variables](/learn/images/tutorial_sending_a_message_to_a_service_define_configurable_variables.gif)
+
+    The source code we generated is as follows.
+
+    ```ballerina
+    configurable int port = 8290;
+    configurable string healthcareBackend = "http://localhost:9090/healthcare";
+    ```
+
+4. Define a record corresponding to the payload from the backend service.
+
+    ```ballerina
+    type Doctor record {|
+        string name;
+        string hospital;
+        string category;
+        string availability;
+        decimal fee;
+    |};
+    ```
+    
+    > **Note:** You can use the "Paste JSON as record" VS Code command to generate the record if you have the JSON payload.
```

Comment:
We discussed using sample JSON payload as the default, right?

So 

4. Generate the record corresponding to the payload from the backend service using the "Paste JSON as record" VS Code command by providing a sample of the expected JSON payload.

At the end we can say, if there is not sample JSON they can manually define the record. Basically, prioritize using the command.